internal/utils/test_utils/test_utils.go

```diff
@@ -0,0 +1,54 @@
+package test_utils
+
+import (
+	"bytes"
+	"github.com/stretchr/testify/require"
+	"io"
+	"log"
+	"testing"
+)
+
+var ConditionalLoggerEnabled bool
+
+func ConditionalLog(v ...any) {
```

Comment:
I was going to suggest more idiomatic way to toggle logging levels, but it seems this is not used anyway, remove?

Reply:
What is that way? I think it's nice to have a tool lying around even if not used on main

Reply:
I'd also remove. Imo having only a single toggle for conditional logger is too wide - imo it would be useful if could toggle individually, e.g. based on environment variable:
```
// ConditionalLog logs v... when environment variable envToggle is defined
func ConditionalLog(envToggle string, v ...any)
```

But I would mainly remove it because it is incompatible with `github.com/consensys/gnark/logger` package. I'd keep a single logging approach so that we would have unified message style etc.